velox/exec/MergingVectorOutput.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/type/Type.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::exec {
+
+class MergingVectorOutput {
+ public:
+  MergingVectorOutput(
```

Comment:
If you want to do both the merging the triggering in one file, you need to pass in a lambda as the triggering criteria to be coming from the sink operator of the pipeline.

Reply:
I also discovered  that if the filter is followed by a limit 1, it does not actually need to merge sparse vectors. But this case can solved by set min_output_batch_rows=1.  

Can you describe the scenario that need adaptivity to handle?

Reply:
There is a trade-off between the cost of merging vectors vs the speed-up we get with a larger batch downstream.  It's not always beneficial to merge the batch to a certain size.  The optimal design actually should be changing the `minOutputBatchRows` adaptively according to the sink operator performance.

Also you need to special treatment whenever it is between a partial limit and final limit, cannot rely on either session property or adaptivity, see https://github.com/facebookincubator/velox/blob/main/velox/exec/LocalPlanner.cpp#L389

Reply:
done

Reply:
done